File: leetcode/java/0338_counting_bits.java
class Solution {
    public int countBit(int n) {
        int numBits = 0;

        do {
            numBits += n % 2;
            n /= 2;
        } while (n > 0);

        return numBits;
    }

    public int[] countBits(int n) {
        int[] res = new int[n + 1];

        for (int i = 0; i <= n; i++) {
            res[i] = countBit(i);
        }

        return res;
    }
}

File: leetcode/cpp/0338_counting_bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans(n + 1, 0);

        for (int i = 0; i <= n; i++) {
            for (int j = i; j >= 1 && i != 0; j /= 2) {
                if (j % 2 != 0) {
                    ans[i] += 1;
                }
            }
        }

        return ans;
    }
};